File: official/guides/getting-started/java/buy_shipment.java
package getting_started;

import java.util.HashMap;
import java.util.List;

import com.easypost.EasyPost;
import com.easypost.exception.EasyPostException;

public class BuyShipment {
    public static void main(String[] args) throws EasyPostException {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");

        List<String> buyCarriers = new ArrayList<String>();
        buyCarriers.add("USPS");
        List<String> buyServices = new ArrayList<String>();
        buyServices.add("First");
        shipment.buy(shipment.lowestRate(buyCarriers, buyServices));

        // OR

        Map<String, Object> rate = new HashMap<String, Object>();
        rate.put("id", "{RATE_ID}");
        shipment.buy(rate);
    }
}

File: official/guides/getting-started/node/buy_shipment.js
// If you already have a saved shipment, or a shipment initialized
// with an id:

shipment.buy(shipment.lowestRate(['USPS'], ['First'])).then(console.log);

// or

shipment.buy('{RATE_ID}').then(console.log);

// If you do not have a saved shipment yet, you must save it first:
shipment.save().then((s) => s.buy(shipment.lowestRate(['USPS'], ['First'])).then(console.log));
